packages/swapper/src/utils.ts

```diff
@@ -179,14 +180,32 @@ export const checkEvmSwapStatus = async ({
   message: string | undefined
 }> => {
   try {
+    const safeTransactionInfo = await fetchSafeTransactionInfo({ chainId, maybeSafeTxHash: txHash })
+    const { isSafeTxHash, transaction } = safeTransactionInfo
+
+    // No buyTxHash handling is correct - we mutate with the actual on-chain transaction, meaning the regular flow then takes over
+    if (
+      isSafeTxHash &&
+      transaction?.confirmations &&
+      Number(transaction.confirmations.length) < transaction.confirmationsRequired
+    ) {
+      return {
+        status: TxStatus.Pending,
+        message: `SAFE proposal submitted. ${transaction.confirmations.length} out of ${transaction.confirmationsRequired} signed.`,
+        buyTxHash: undefined,
+      }
+    } else if (transaction?.transactionHash) {
+      // Mutate with  the actual on-chain transaction work and let things work as-is
+      txHash = transaction.transactionHash
+    }
```

Comment:
same logic as thorchain, maybe chance to lift out into a helper func?

Reply:
Consolidated in 00689770c10109101fb4a4b1c153a51ace2a5e94 and added to Li.Fi/CoW while at it. Was already explicitly implemented for THOR, and implicitly with Arb Bridge/Portals/1inch/ZRX.

Retested with:

- 0x: https://jam.dev/c/90da8c11-05d9-4b56-8251-8f39661e2730
- Portals: https://jam.dev/c/bec70bb8-e3f9-483e-ba23-260d6b75d080
- THOR: https://jam.dev/c/ebc9aeb4-bace-4d0f-bfd7-b440944c112b
- Li.Fi: https://jam.dev/c/75d98219-c493-4e97-8d65-25cdbf9250ed
- Li.Fi - cross-chain swap: https://jam.dev/c/b28c4006-d585-4ec2-a5b1-042147257758 (note: the unexpected error is also happening on develop with regulr, non sc swaps during polling)
- THOR cross-chain EVM -> RUNE swap: https://jam.dev/c/6e894023-cc8f-4e5c-a29f-c29ca897a507
- CoW: Confirmed sad as it wcv2 hdwallet doesn't have `ethSignTypedData` implemented yet - captured in https://github.com/shapeshift/web/issues/7272:
  <img width="400" alt="image" src="https://github.com/user-attachments/assets/fa263a00-5741-45cf-9152-b8886fc785bd">

Reply:
Small cleanup: https://github.com/shapeshift/web/pull/7506/commits/122e507f3b94674c735322150762883810412708

validated:
- 0x on avax
- lifi on avax (same chain)
- thorchain on avax (same chain)